### PR TITLE
feat: Auth

### PR DESCRIPTION
**Pull request by Shivani-1524**
Added Auth and private routing.

login credentials: 
email - test@email.com
password -  test

files to be reviewed:
App.js
AuthPages Folder
Navbar.js
RequiresAuth.js

### CODE REVIEW

**Review by sri-rishi — APPROVED** [review]
Nice work, code and UI looks good, just a remainder after done working , remove `console.log` from code

File: src/Pages/AuthPages/authpage.css [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    height: 100vh;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    background-color: var(--violet-color-light);
+  }
+  
+  .auth-card {
+    margin-top: 60px;
```
Comment (Sahil25061999) [inline src/Pages/AuthPages/authpage.css:12]: you can use rem here.

File: src/Pages/HomePage/HomeComponent/CategoryChip.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React, { useState } from 'react'
+import { useCategory } from "../../../Context/category-provider"
+
+const CategoryChip = ({ props }) => {
+    const { title } = props
+    // const [setCategorySelect, categorySelect] = useState(false);
```
Comment (Sahil25061999) [inline src/Pages/HomePage/HomeComponent/CategoryChip.js:6]: remove comments if not needed.

**Review by Sahil25061999 — APPROVED** [review]
Looks good just suggested some minor changes related to rem-px conversion, beautiful use of icon in the navbar.

**Pull request merged**